src/hotspot/os/posix/os_posix.cpp

```diff
@@ -706,7 +709,26 @@ void* os::dll_lookup(void* handle, const char* name) {
 }
 
 void os::dll_unload(void *lib) {
-  ::dlclose(lib);
+#if defined(LINUX)
+  struct link_map *lmap;
+  int res_dli = ::dlinfo(lib, RTLD_DI_LINKMAP, &lmap);
+  const char* l_path = "<none>";
+  if (res_dli == 0) {
+    l_path = lmap->l_name;
+  }
```

Comment:
Can we hide this in `os::Linux::dll_path` so that the code here would simply be:

```
const char* l_path = LINUX_ONLY(os::Linux::dll_path(lib))
                     NOT_LINUX("<not available>");

```